services/bundle_analysis.py

```diff
@@ -507,6 +507,45 @@ def bundle_analysis_loader(self):
     def bundle_report(self) -> Optional[BundleAnalysisReport]:
         return self.bundle_analysis_loader.load(self.commit_report.external_id)
 
+    def _has_required_changes(
+        self, comparison: BundleAnalysisComparison, pull: EnrichedPull
+    ) -> bool:
+        """Verifies if the notifier should notify according to the configured required_changes.
+        Changes are defined by the user in UserYaml. Default is "always notify".
+        Required changes are bypassed if a comment already exists, because we should update the comment.
+        """
+        if pull.database_pull.bundle_analysis_commentid:
+            log.info(
+                "Skipping required_changes verification because comment already exists",
+                extra=dict(pullid=pull.database_pull.id, commitid=self.commit.commitid),
+            )
+            return True
+
+        required_changes: bool | Literal["bundle_increase"] = read_yaml_field(
+            self.current_yaml, ("comment", "require_bundle_changes"), False
+        )
+        changes_threshold: int = read_yaml_field(
+            self.current_yaml, ("comment", "bundle_change_threshold"), 0
+        )
+        match required_changes:
+            case False:
+                return True
+            case True:
+                return abs(comparison.total_size_delta) > changes_threshold
+            case "bundle_increase":
+                return (
+                    comparison.total_size_delta > 0
+                    and comparison.total_size_delta > changes_threshold
+                )
+            case _:
+                log.warning(
```

Comment:
This is technically dead code because the YAML validation should let it get this far.
I added this just so it wouldn't break the interface by returning None... maybe I should remove it? Even if it does return `None` it won't break the app.

Reply:
I think it's fine as long as you leave a comment explaining that this code should _never_ be reached, (maybe even throw an exception? emit a metric? not sure, but we would need some way to observe that this does not happen)